include/tensor.hpp

```diff
@@ -0,0 +1,30 @@
+#include <vector>
+#include <memory>
+
+namespace deepworks {
+
+using Shape = std::vector<size_t>;
+using Strides = std::vector<size_t>;
+
+class Tensor {
+public:
+    using Type = float;
+    explicit Tensor(const Shape &shape);
+    Tensor();
+
+    Tensor(const Tensor&) = default;
```

Comment:
`Tensor(const Tensor&)`
`Tensor(Tensor&&)` 

generated by compiler, there is no necessary to put `default`